src/contexts/products-context.js

```diff
@@ -0,0 +1,47 @@
+import { useReducer, useContext, createContext } from "react";
+
+const ProductContext = createContext();
+
+function reducerFunc(prevState, action) {
+  switch (action.TYPE) {
+    case "SHOW_ONLY_IN_STOCK": {
```

Comment:
Suggestion : Moving all reducer functions to seperate files

Reply:
Noted!